crates/miden-objects/src/transaction/chain_mmr.rs

```diff
@@ -64,9 +67,55 @@ impl ChainMmr {
         Ok(Self { mmr, blocks: block_map })
     }
 
+    /// Converts the [`Mmr`] into a [`ChainMmr`] by selectively copying all leaves that are in the
+    /// given `blocks` iterator.
+    ///
+    /// This tracks all blocks in the given iterator in the [`ChainMmr`] except for the block whose
+    /// block number equals [`Mmr::forest`], which is the current chain length.
+    ///
+    /// # Panics
+    ///
+    /// Due to being only available in test scenarios, this function panics when one of the given
+    /// blocks does not exist in the provided mmr.
+    #[cfg(any(feature = "testing", test))]
+    pub fn from_mmr<I>(
+        mmr: &Mmr,
+        blocks: impl IntoIterator<Item = BlockHeader, IntoIter = I> + Clone,
+    ) -> Result<ChainMmr, ChainMmrError>
```

Comment:
We usually put such code into the `testing` module (e.g., under `src/testing/chain_mmr.rs`).